test/functional/plugin/lsp_spec.lua

```diff
@@ -817,6 +817,87 @@ describe('LSP', function()
     end)
   end)
 
+  describe('apply_text_document_edit', function()
+    local target_bufnr
+    before_each(function()
+      target_bufnr = exec_lua [[
+        local bufnr = vim.fn.bufadd("fake/uri")
+        local lines = {"1st line of text", "2nd line of text"}
```

Comment:
Please add a non-ASCII character to one of the lines (maybe "Ä 2nd line of text"?) to test for encoding.

Reply:
don't. Let's keep meaningful commits, non-ASCII characters should be merged with the same commits that introduce supports for them i.e., https://github.com/neovim/neovim/pull/12223

Reply:
No, that commit didn't introduce support, non-ASCII characters should have worked from the start, so it fixes a bug this test would have caught.

Reply:
(But once the test is merged, I can try to extend it with non-ASCII characters. I'd prefer not to merge this PR into my PR, though.)